components/teams-service/storage/postgres/postgres.go

```diff
@@ -258,42 +258,44 @@ func (p *postgres) GetTeams(ctx context.Context) ([]storage.Team, error) {
 // server code before we get here if the team in question was filtered, but if we call this
 // from a different context we should apply project filtering in this function as well.
 // Not doing it to save us a database call since it's not needed currently.
-func (p *postgres) RemoveUsers(ctx context.Context, id string, userIDs []string) (storage.Team, error) {
+func (p *postgres) RemoveUsers(ctx context.Context, id string, userIDs []string) (updatedUserIDs []string, err error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	tx, err := p.db.BeginTx(ctx, nil)
+	projectsFilter, err := ProjectsListFromContext(ctx)
 	if err != nil {
-		return storage.Team{}, p.processError(err)
+		return nil, p.processError(err)
 	}
 
-	res, err := tx.ExecContext(ctx,
-		`DELETE FROM teams_users_associations
-		WHERE user_id = ANY($1) AND team_db_id = team_db_id($2)`,
-		pq.Array(userIDs), id)
-	if err != nil {
-		return storage.Team{}, p.processError(err)
-	}
-	num, err := res.RowsAffected()
+	var dbID int
+	var teamProjects []string
+	// ensure the team exists and isn't filtered out by the project filter
+	err = p.db.QueryRowContext(ctx,
```

Comment:
could be a candidate for a helper function since I used this across AddUsers, RemoveUsers, and GetTeamForUsers

Reply:
We should start and do all of these related queries in a transaction so things don't change from under us.

Reply:
Oh good point, I missed adding the transaction here